tools/config/__init__.py

```diff
@@ -1058,9 +1062,29 @@ def validate_config(self):
 
         Arguments: None
         """
+
         params, _ = self.get_config_data()
+        err_msg = ""
+
+        for param in params:
+            min      = params[param].value_min
+            max      = params[param].value_max
+            accepted = params[param].accepted_values
+            value    = params[param].value
+
+            if (value < min or (value > max if max is not None else False)):
+                err_msg += "\nInvalid config range for %s, value %s is not in the required range: [%d:%d]"\
+                                % (param, value, min if min is not None else -1, max if max is not None else -1)
+
+            if (value not in accepted if accepted else False):
```

Comment:
I would write this as
```suggestion
            if accepted and value not in accepted:
```